### DISCUSSION TITLE
Techniques to manage input schemas for different mutations (create | update)?

### DISCUSSION THREAD

**Discussion by laneme** [MichalLytek/type-graphql#1305] [Q&A]
First of all, guys, thank you all for maintaining such an amazing and useful repository. 
Second, i'm really new to both graphql and type-graphql. So I may be lacking certain common knowledge. 

Consider this input class where `displayName` is a required field.
```ts
@InputType()
export class UserInput {
    @Field()
    displayName!: string

    @Field()
    profilePhoto?: string
}
```

This works well for my create mutation. But if I were to implement an update mutation, where i would like to make all fields optional, i need to create another input type like `UserInputForUpdate`
```ts
@InputType()
export class UserInputForUpdate {
    @Field()
    displayName?: string

    @Field()
    profilePhoto?: string
}
```

This essentially adds duplication across all of my schemas. Which sometimes I do not mind but still brings me to a couple of questions
- Is there a way i can make all UserInput fields optional
- How does the community do this (write input schemas for different type of mutations)

One thanks in advance ;)

**Comment by MichalLytek** [comment]
See #453 👀
